Library/OcCpuLib/OcCpuLib.c

```diff
@@ -913,8 +915,20 @@ ScanAmdProcessor (
     Cpu->CurBusRatio = Cpu->MaxBusRatio;
     Cpu->MinBusRatio = Cpu->MaxBusRatio;
 
-    Cpu->FSBFrequency = DivU64x32 (Cpu->CPUFrequency, Cpu->MaxBusRatio);
-  }
+    if (Cpu->CPUFrequency == 0) {
+	    //
+	    // There may be some quirks with virtual CPUs (VMware is fine).
+	    // Formerly we checked Cpu->MinBusRatio > 0, but we have no MinBusRatio on Penryn.
+	    //
+        if (Cpu->CPUFrequencyFromTSC > 0 && Cpu->MaxBusRatio > Cpu->MinBusRatio) {
```

Comment:
FSBFrequency calculation does not differ inside and outside if this code, and the comment above mentions Penryn, which is Intel, not AMD. The change does not make sense to me.